guides/updating/versions/0.2-to-0.3.md

```diff
@@ -0,0 +1,636 @@
+4. Some indy objects, as the preview should now be passed only as their attributes (i.e. no need of creating the object instance) and provided in the `proofFormats.indy` object.
+
+<!--tabs-->
+
+##### 0.2.x
+
+```ts
+await agent.proofs.proposeProof(
+      'connectionId',
+      new PresentationPreview({
+        attributes: [new PresentationPreviewAttribute({ name: 'key', value: 'value' })],
+        predicates: [
+          new PresentationPreviewPredicate({
+            name: 'age',
+            credentialDefinitionId,
+            predicate: PredicateType.GreaterThanOrEqualTo,
+            threshold: 50,
+          }),
+        ],
+      })
+    )
+```
+
+##### 0.3.x
+
+```ts
+await agent.proofs.proposeProof({
+      connectionId: connection.id, 
+      protocolVersion: 'v1',
+      proofFormats: {
+        indy: {
+          attributes: [{ name: 'key', value: 'value' }],
+          predicates: [{name: 'age', credentialDefinitionId, predicate: PredicateType.GreaterThanOrEqualTo, threshold: 50, ]
+        },
+      },
+      comment: 'Propose proof comment',
+    })
+```
+
+<!--/tabs-->
+
+#### Data from Received Messages only Stored in Record after Accepting
+
+Previously when we received a message from another connection we would store the relevant data from the exchange in the `ProofRecord`. The values we would store were the `credentialDefinitionId` and `schemaId` in the credential metadata, and the `credentialAttributes` field.
+
+Starting with AFJ 0.3.0 the values are not stored in the proof record until after the message content has been accepted (in the case of an offer this means after sending the request message). This is to avoid ambiguity of the values in the credential record. If I have sent a proposal and then receive an offer, should the credential record contain the values from the proposal or the values from the offer? The first one reflects our view on what the data should be, the second one reflects the latest data.
+
+We decided to make the record properties always hold our view of what the data should be, and only update it after accepting the contents of a received message (either using auto accept, or by calling the `acceptXXX` methods on the credential module).
+
+This is an important change and requires some updates to how you extract the relevant data from the offer (or other messages such the proposal). We've added a new `getFormatData` method on the Proofs API that allows you to retrieve the attributes and format data for all messages in an exchange. One of the advantages of this approach is that we don't have to store all relevant data in the ProofExchangeRecord anymore, which helps when adding new formats that don't match with the attributes used for indy credentials. In addition, the return value for this method is the same whether v1 or v2 of the protocol is used. This means your code should only care about the proof format (indy in this case) and doesn't have to worry about the protocol version.
```

Comment:
I thnk we don't store any of these properties in a proof record, so it doesn't apply for proofs, but would be good to have the getFormatData example to showcase how you can access attachment data